### DISCUSSION TITLE
Reprojecting to UTM breaks everything

### DISCUSSION THREAD

**Discussion by NobodyLivesHere** [earthlab-education/Earth-Analytics-2023-01-Intro#687] [! GitHub help]
HI all, when going back to harmonize my data and run my fuzzy rules for the DEM and slope sets, I started encountering massive problems when switching from a PlantCarreee projection (WGS 84) to (UTM 15N).  When sticking to lat and lon, all my downloader functions were working and everyting was clipping and plotting correctly.  After projecting, it spits out a valid gdf with bounds for the study site but starts erroring out when trying to plot.

![image](https://github.com/earthlab-education/Earth-Analytics-2023-01-Intro/assets/88915377/366bf8dc-ba19-42d2-a65f-0ff5ccfed5c9)

![image](https://github.com/earthlab-education/Earth-Analytics-2023-01-Intro/assets/88915377/38412177-b4df-42bc-9efe-e34abe239c27)

![image](https://github.com/earthlab-education/Earth-Analytics-2023-01-Intro/assets/88915377/91c16a22-e883-4c13-9511-81913903686f)

I thought the problem was confined to just one grassland and maybe it was because the location was too close to the edges of the Mercator but the problem persists no matter if I change either the study site or the UTM.  Even using a generic Mercator projection causes an issue.  After further inspection, I noticed that grassland gdf was originally importing as NAD83, but even using the appropriate Mercator projections for that do not fix the issue.  

For now I may have to exclude the DEM and slope data from the final analysis since time is running low, but I'm wondering if anyone else had the same issue?

**Comment by edchanonline** [comment]
You have to use the `ds.rio.reproject_match()`. I reprojected everything to elevation data but I suspect you have to reproject match your data arrays instead of just reprojecting to the utm espg code, which took me a good while to figure out. Hope this helps!

**Reply by NobodyLivesHere** [reply]
The code is set to reproject_match to harmonize all the arrays, the problem is that I can't get it into Mercator in the first place without causing major distortion that makes the data unable to be parsed.

**Comment by eculler** [comment]
Hm, do you still have this issue when plotting with .plot()? It may be a geoviews problem, as it doesn't always handle the CRS correctly unfortunately.

No worries about excluding slope from the final analysis at this point (or keeping everything in geographic coordinates for the final analysis, for that matter -- it's only the slope calculation that is affected by units of degrees.)

**Reply by NobodyLivesHere** [reply]
Yeah I still have no clue what was going on, I just ended up submitting what I had and noting what the issue was extensively in the documentation.  Basically reprojecting into Mercator was breaking _any_ use o fthe datasets.  They wouldn't plot with hvplot. The individual pixel values were going insane in ways that hardly made any sense..  It was not reading the bounds correctly either and couldnt parse through even fetching the data without erroring out.  

The good news is that the fuzzy model does work as intended when passed clean test data and when testing with individual sets such as soil in NAD83.  It should be then possible to do a clean run by taking out a few lines from the settings json and editing a couple in the jupyter notebook to exclude slope.  I also needed to get around to writing in NaN handling as that seemed to be affecting calculation output.
